src/main/java/com/antkorwin/betterstrings/ast/ExpressionParser.java

```diff
@@ -45,24 +45,21 @@ private CompilationUnitTree getCompilationUnitTree(String code) {
 		                                                null,
 		                                                null,
 		                                                null,
-		                                                Arrays.asList(new FakeJavaFileWrapper(code)));
-
-		try {
-			return ct.parse().iterator().next();
-		} catch (IOException e) {
-			e.printStackTrace();
-			throw new RuntimeException("Error while parsing expression in the string literal: " + code, e);
-		}
+		                                                Collections.singletonList(new FakeJavaFileWrapper(code)));
+		return ct.parse().iterator().next();
 	}
 
-
 	private class FakeJavaFileWrapper extends SimpleJavaFileObject {
 
-		private String text;
+		private final String text;
 
 		public FakeJavaFileWrapper(String text) {
 			super(URI.create("myfake:/Test.java"), JavaFileObject.Kind.SOURCE);
-			this.text = "class Test { Object value = String.valueOf(" + text + "); }";
+			if (callToStringExplicitlyInInterpolations) {
+				this.text = "class Test { Object value = java.util.Objects.nonNull(" + text + ") ? java.util.Objects.requireNonNull(" + text + ").toString() : \"null\"; }";
```

Comment:
That's the best I've come up with so far, it performs boxing for primitive types, but unfortunately requires double evaluation of the expression :/